Source/Project64/UserInterface/SettingsConfig.cpp

```diff
@@ -251,9 +251,9 @@ void CSettingConfig::ApplySettings(bool UpdateScreen)
     if (!GameIni.empty())
     {
         stdstr GoodName;
-        if (!g_Settings->LoadStringVal(Game_GoodName, GoodName))
+        if (!g_Settings->LoadStringVal(Cfg_GoodName, GoodName))
```

Comment:
the way the setting class works is that it will load it from the source, if it does not exist it will read if from the default.

it will return true if read from the source, false if from the default.

So if this returned false before it read it from rdb_goodname so had the goodname now in the variable, and then could just write it.

Looks good, so basically all reads come from rdb_goodname and all writes are to cfg_goodname